### PR TITLE
APP-2975 Fixed token scopes validation, github

### PR DESCRIPTION
**Pull request by josecordaz**

### CODE REVIEW

File: integrations/github/validate.go [hunk 1]
```diff
@@ -37,20 +37,33 @@ func (s *Integration) checkTokenScopes() error {
...
+	if !m["read:org"] {
 		return scopeErr("read:org")
 	}
-	if !m["repo"] {
-		return scopeErr("repo")
+	{
+		if !m["repo:status"] {
+			return scopeErr("repo:status")
+		}
+		if !m["repo_deployment"] {
```
Comment (jhaynie) [inline integrations/github/validate.go]: we don't pull deployments at the moment so we shouldn't need this.

File: integrations/github/validate.go [hunk 2]
```diff
@@ -37,20 +37,33 @@ func (s *Integration) checkTokenScopes() error {
...
+			return scopeErr("repo_deployment")
+		}
+		if !m["public_repo"] {
+			return scopeErr("public_repo")
+		}
+		if !m["repo:invite"] {
```
Comment (jhaynie) [inline integrations/github/validate.go]: we don't need to do anything with repo invitations so we can remove this.

File: integrations/github/validate.go [hunk 3]
```diff
@@ -37,20 +37,33 @@ func (s *Integration) checkTokenScopes() error {
...
-	if !m["repo"] {
-		return scopeErr("repo")
+	{
+		if !m["repo:status"] {
+			return scopeErr("repo:status")
+		}
+		if !m["repo_deployment"] {
+			return scopeErr("repo_deployment")
+		}
+		if !m["public_repo"] {
```
Comment (jhaynie) [inline integrations/github/validate.go]: not sure we need this. we should test w/o it based on https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/

**Comment by xzandrew** [comment]
This change removes the required parent scopes (m["repo"]), but last time I checked these were required. Have you tested with ripsrc export as well?

**Comment by josecordaz** [comment]
> This change removes required parent scopes (m["repo"]), but last time I checked export wasn't working without them. Have you tested export including git processing?

I will test it

**Comment by xzandrew** [comment]
I don't think we should make this change, it's better to return an error when there is not access to private repos that proceed and print warning.

**Pull request closed without merging**